osmpbf/decode_data.go

```diff
@@ -513,6 +519,38 @@ func (dec *dataDecoder) scanWays(data []byte) error {
 				way.Nodes[index].ID = osm.NodeID(prev)
 				index++
 			}
+		case 9: // lat
+			dec.wlats, err = msg.Iterator(dec.wlats)
+			if err != nil {
+				return err
+			}
+
+			var prev, index int64
+			for dec.wlats.HasNext() {
```

Comment:
need to preallocate way.Nodes here

Reply:
OK, I'll push another commit (paying attention not to get rid of the node IDs).